cartography/intel/gcp/pubsub.py

```diff
@@ -0,0 +1,232 @@
+import json
+import logging
+from typing import Dict
+from typing import List
+
+import time
+import neo4j
+from googleapiclient.discovery import HttpError
+from googleapiclient.discovery import Resource
+
+from cartography.util import run_cleanup_job
+from . import label
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_pubsub_subscriptions(pubsub: Resource, project_id: str, regions: list, common_job_parameters) -> List[Dict]:
+    subscriptions = []
+    try:
+        req = pubsub.projects().subscriptions().list(project=f"projects/{project_id}")
+        while req is not None:
+            res = req.execute()
+            if res.get('subscriptions'):
+                for subscription in res['subscriptions']:
+                    subscription['project_id'] = project_id
+                    subscription['region'] = 'global'
+                    subscription['id'] = subscription['name']
+                    subscription['subscription_name'] = subscription.get('name').split('/')[-1]
+                    subscriptions.append(subscription)
+            req = pubsub.projects().subscriptions().list_next(previous_request=req, previous_response=res)
+
+        if common_job_parameters.get('pagination', {}).get('pubsub', None):
```

Comment:
using the same pagination variable for both topic & subscription may not help. we have to re-think this.

Also, i think topics could be many. But, when it comes to subscriptions, they are always limited.